lib/features/family/features/home_screen/widgets/daily_experience_container.dart

```diff
@@ -0,0 +1,118 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:font_awesome_flutter/font_awesome_flutter.dart';
+import 'package:givt_app/features/family/app/injection.dart';
+import 'package:givt_app/features/family/features/home_screen/cubit/daily_experience_cubit.dart';
+import 'package:givt_app/features/family/features/home_screen/widgets/models/daily_experience_custom.dart';
+import 'package:givt_app/features/family/shared/design/components/content/fun_progressbar.dart';
+import 'package:givt_app/features/family/shared/widgets/texts/label_large_text.dart';
+import 'package:givt_app/features/family/utils/family_app_theme.dart';
+import 'package:givt_app/shared/widgets/base/base_state_consumer.dart';
+
+class DailyExperienceContainer extends StatefulWidget {
+  const DailyExperienceContainer({super.key});
+
+  @override
+  State<DailyExperienceContainer> createState() =>
+      _DailyExperienceContainerState();
+}
+
+class _DailyExperienceContainerState extends State<DailyExperienceContainer> {
+  final DailyExperienceCubit _cubit = getIt<DailyExperienceCubit>();
+
+  Timer? _timer;
+  late int _minutesUntilReset;
+
+  @override
+  void initState() {
+    super.initState();
+    _cubit.init();
+  }
+
+  @override
+  void dispose() {
+    _timer?.cancel();
+    _cubit.close();
+    super.dispose();
+  }
+
+  void _startCountdown(DateTime timeOfReset) {
+    _timer?.cancel();
+    _minutesUntilReset = timeOfReset.difference(DateTime.now()).inMinutes;
+    _timer = Timer.periodic(const Duration(minutes: 1), (Timer timer) {
```

Comment:
Is this something the UI should do or the cubit? (cubit can trigger a refresh of the UI)

Reply:
UI because it can then redraw a smaller part.